File: reference/src/ir/visibility.ts
import type { RuleGraph } from "./graph.js";
import type { TangleDiagnostic } from "../model.js";

// Check that all symbols referenced in the IR are visible (exported or module-internal)
export function checkIRVisibility(graph: RuleGraph, exportedSymbols: Set<string>): TangleDiagnostic[] {
  const diags: TangleDiagnostic[] = [];
  // For now, all nodes pass visibility check since cross-module references aren't yet tracked
  // Future: check that each IRNode's label references only visible symbols
  return diags;
}

File: compiler/tangle-cli/src/ir/visibility.rs
use crate::ir::graph::RuleGraph;
use crate::model::TangleDiagnostic;

pub fn check_ir_visibility(_graph: &RuleGraph, _exported_symbols: &[String]) -> Vec<TangleDiagnostic> {
    // Placeholder — cross-module reference validation
    vec![]
}
